warehouse/cache/origin/derivers.py

```diff
@@ -0,0 +1,38 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+from warehouse.cache.origin.interfaces import IOriginCache
+
+
+def html_cache_deriver(view, info):
+    renderer = info.options.get('renderer')
+    if renderer and renderer.name.endswith('.html'):
+        def wrapper_view(context, request):
+            try:
+                cacher = request.find_service(IOriginCache)
+            except ValueError:
+                pass
+            else:
+                request.add_response_callback(
+                    functools.partial(
+                        cacher.cache,
+                        sorted(['all-html', renderer.name]),
+                        seconds=1 * 24 * 60 * 60,                 # 1 day
```

Comment:
This is probably wrong? We likely don't want to add a cache like this for every single rendered view automatically (and I'm not sure offhand what the behavior would be if something already declared a cache, which one would "win").

Reply:
OK, I can see how this would break views that already have an `origin_cache` decorator. I'll think about that.

Maybe I'm misinterpreting #778 -- isn't the goal to have a surrogate key for every page rendered from a given template? Maybe this isn't really necessary if we can easily handle a  purge of `all-html` instead? 

Are you saying that automatically adding `all-html` doesn't make sense here either?

Reply:
Or are you saying that we _only_ need to add the `Surrogate-Key` header, and shouldn't add these `Surrogate-Control` values?

Reply:
@di Adding the surrogate key is fine, it's the automatic addition of a ``CacheControl`` header to every response that I think is the wrong thing to do here. If it doesn't support it already, you might have to make the cache control stuff handle ``seconds=None`` etc.

As you mentioned, you'll also need to figure out a solution for things that already have a ``origin_cache`` header as well.